Update advanced plugin deployment docs for jenkinsci/jenkins#5862 (#4677)
jenkins-infra/jenkins.io: content/doc/book/managing/plugins.adoc

plugins which have been developed and maintained by various members of the
Jenkins community.

This section covers everything from the basics of managing plugins within
the Jenkins web UI, to making changes on the <<../glossary#controller,controller's>>
file system.

== Installing a plugin

> Jenkins provides two different methods for installing plugins on the controller:

Author: ```suggestion
Jenkins provides two methods for installing plugins on the controller:
```

Jenkins community.

> This section will cover everything from the basics of managing plugins within

Reviewer: ```suggestion
This section covers everything from the basics of managing plugins within
```

the Jenkins web UI, to making changes on the <<../glossary#controller,controller's>>
file system.

== Installing a plugin

> Jenkins provides a couple of different methods for installing plugins on the controller:

Reviewer: ```suggestion
Jenkins provides two different methods for installing plugins on the controller:
```

[CAUTION]
====

> If the list of available plugins is empty, the controller might be incorrectly

Reviewer: ```suggestion
If the list of available plugins is empty, the controller may be incorrectly
```

configured or has not yet downloaded plugin meta-data from the Update Center.

> Clicking the *Check now* button will force Jenkins to attempt to contact its

Reviewer: ```suggestion
Clicking the *Check now* button forces Jenkins to attempt to contact its
```

. Navigate to the *Manage Jenkins* > *Manage Plugins* page in the web UI.
. Click on the *Advanced* tab.
. Choose the `.hpi` file from your system or enter a URL to the `.hpi` file under the *Deploy Plugin* section.

> . *Depoloy* the plugin file.

Reviewer 2: ```suggestion
. *Deploy* the plugin file.
```

image::managing/plugin-manager-upload.png["Advanced tab in the Plugin Manager", role=center]

Once a plugin file has been uploaded, the Jenkins controller must be manually
restarted in order for the changes to take effect.

==== On the controller

> Assuming a `.hpi` file has been explicitly downloaded by a systems

Reviewer: ```suggestion
Assuming a `.hpi` file has been explicitly downloaded by a system
```

the Jenkins controller (for example, on Debian systems `JENKINS_HOME` is generally
`/var/lib/jenkins`).

> The controller will need to be restarted before the plugin is loaded and

Reviewer: ```suggestion
The controller must be restarted before the plugin is loaded and
```
